### ISSUE TITLE
MCM won't work if vehicle ID is different from 1

### ISSUE THREAD

**Issue by rafaellehmkuhl** [bluerobotics/BlueOS#2396]

**Comment by joaoantoniocardoso** [comment]
Could you elaborate?

**Comment by rafaellehmkuhl** [comment]
If the user deploys BlueOS on a vehicle and changes the Ardupilot ID to a number other than 1, this won't be passed to the MCM command line.

This is purely a theory, looking at the code and discussing with the rest of the team thought. I didn't test it.

**Comment by joaoantoniocardoso** [comment]
Right. MCM is now relying on the environment variable, which cannot be dynamically changed.

**Comment by rafaellehmkuhl** [comment]
Perfect. So it's a bug indeed.

I've opened it because there are some people already trying blueos+cockpit with a fleet of vehicles, and so we are adding support for it on Cockpit, but it would only work for telemetry as of now.

**Comment by joaoantoniocardoso** [comment]
BTW, if the person uses Cockpit, the Mavlink ID for the MCM would be irrelevant.

**Comment by rafaellehmkuhl** [comment]
> BTW, if the person uses Cockpit, the Mavlink ID for the MCM would be irrelevant.

Humnnnn, true!
We don't check it.
It would only be relevant for QGC for example, that relies on the ID, right?

**Comment by joaoantoniocardoso** [comment]
This is actually already solved: we pass `$MAV_SYSTEM_ID` already expanded, so MCM receives the number directly.
